File: VB/ViewModel.vb
Imports System.Collections.ObjectModel

Namespace HighlightModifiedCells

    Public Class ViewModel

        Public Property Customers As ObservableCollection(Of Customer)

        Public Sub New()
            Customers = New ObservableCollection(Of Customer)()
            For i As Integer = 1 To 30 - 1
                Customers.Add(New Customer() With {.ID = i, .Name = "Customer" & i, .RegistrationDate = Date.Now.AddDays(i)})
            Next
        End Sub
    End Class

    Public Class Customer

        Public Property ID As Integer

        Public Property Name As String

        Public Property RegistrationDate As Date
    End Class
End Namespace

File: CS/ViewModel.cs
using System;
using System.Collections.ObjectModel;

namespace HighlightModifiedCells {
    public class ViewModel {
        public ObservableCollection<Customer> Customers { get; set; }
        public ViewModel() {
            Customers = new ObservableCollection<Customer>();
            for (int i = 1; i < 30; i++) {
                Customers.Add(new Customer() { ID = i, Name = "Customer" + i, RegistrationDate = DateTime.Now.AddDays(i) });
            }
        }
    }

    public class Customer {
        public int ID { get; set; }
        public string Name { get; set; }
        public DateTime RegistrationDate { get; set; }
    }
}
